include/kernels/contraction.cuh

```diff
@@ -8,14 +8,490 @@
 
 namespace quda
 {
+  static constexpr int max_contract_results = 16; // sized for nSpin**2 = 16
+ 
+  using spinor_array = array<array<double, 2>, max_contract_results>;
+  using staggered_spinor_array = array<double, 2>; 
+
+  template <typename real, int nSpin = 4> class DRGammaMatrix {
+  public:
+    // Stores gamma matrix column index for non-zero complex value.
+    // This is shared by g5gm, gmg5.
+    int gm_i[nSpin*nSpin][nSpin] {};
```

Comment:
Are the curly braces here to tell the default constructor to zero this array?  If so it should be written as `int gm_i[nSpin*nSpin][nSpin] = {}'`.  Also should there be equivalent for g5gm_z as well then?

Reply:
fixed in 84851f3